.github/workflows/master.yml

```
@@ -15,86 +15,42 @@ permissions:
 
-   linux:
-      needs: api_check
-      runs-on: ubuntu-latest
+   validate-all:
+      name: Run all Gradle validations
       if: github.repository == 'kotest/kotest'
+      needs: check-abi
       strategy:
-         fail-fast: false
          matrix:
-            target:
-               - jvmTest publishJvmPublicationToDeployRepository publishKotlinMultiplatformPublicationToDeployRepository publishKotestBomPublicationToDeployRepository
-               - jsTest publishJsPublicationToDeployRepository
-               - wasmJsTest publishWasmJsPublicationToDeployRepository
-               - linuxX64Test publishLinuxX64PublicationToDeployRepository
-               - linuxArm64TestKlibrary publishLinuxArm64PublicationToDeployRepository
-               - androidNativeX86TestKlibrary androidNativeX64TestKlibrary publishAndroidNativeX64PublicationToDeployRepository publishAndroidNativeX86PublicationToDeployRepository
-               - androidNativeArm32TestKlibrary androidNativeArm64TestKlibrary publishAndroidNativeArm32PublicationToDeployRepository publishAndroidNativeArm64PublicationToDeployRepository
-      steps:
-         -  uses: kotest/kotest-action@master
-            with:
-               gradle-command: ${{ matrix.target }}
-               java-version: "17"
-
-   mac:
-      needs: api_check
-      runs-on: macos-latest
-      if: github.repository == 'kotest/kotest'
-      strategy:
+            include:
+               -  os: macos-latest
+               -  os: ubuntu-latest
+               -  os: windows-latest
          fail-fast: false
-         matrix:
-            target:
-               - macosX64Test publishMacosX64PublicationToDeployRepository
-               - macosArm64Test publishMacosArm64PublicationToDeployRepository
-               - iosX64Test publishIosX64PublicationToDeployRepository
-               - iosSimulatorArm64Test publishIosSimulatorArm64PublicationToDeployRepository
-               - iosArm64TestKlibrary publishIosArm64PublicationToDeployRepository
-               - tvosX64Test publishTvosX64PublicationToDeployRepository
-               - tvosArm64TestKlibrary publishTvosArm64PublicationToDeployRepository
-               - watchosArm32TestKlibrary publishWatchosArm32PublicationToDeployRepository
-               - watchosArm64TestKlibrary publishWatchosArm64PublicationToDeployRepository
-               - watchosX64Test publishWatchosX64PublicationToDeployRepository
-               - watchosSimulatorArm64Test publishWatchosSimulatorArm64PublicationToDeployRepository
-               - tvosSimulatorArm64Test publishTvosSimulatorArm64PublicationToDeployRepository
-      steps:
-         -  uses: kotest/kotest-action@master
-            with:
-               gradle-command: ${{ matrix.target }}
-               java-version: "17"
+      uses: ./.github/workflows/run-gradle.yml
+      with:
+         runs-on: ${{ matrix.os }}
+         ref: ${{ inputs.ref }}
+         task: check --continue
```

```suggestion
         task: check --continue
```

After applying `--continue`, will the build step still fail if some test fails?

---

Yes, the build step will still fail, but Gradle will run all tests in all subprojecs.